cmd/podman/cp.go

```diff
@@ -0,0 +1,255 @@
+
+		} else {
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(mountPoint, ctr.WorkingDir()), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", destPath)
+			}
+			destPath = filepath.Join(mountPoint, ctr.WorkingDir(), destPath)
+		}
+	} else {
+		if filepath.IsAbs(srcPath) {
+			srcPath = filepath.Join(mountPoint, srcPath)
+		} else {
+			srcPath = filepath.Join(mountPoint, ctr.WorkingDir(), srcPath)
+		}
+	}
+	glob, _ = filepath.Glob(srcPath)
+	if len(glob) == 0 {
+		glob = append(glob, srcPath)
+	}
+	if !filepath.IsAbs(destPath) {
+		dir, err := os.Getwd()
+		if err != nil {
+			return errors.Wrapf(err, "err getting current working directory")
+		}
+		destPath = filepath.Join(dir, destPath)
+	}
+
+	var lastError error
+	for _, src := range glob {
+		err := copy(src, destPath, dest, idMappingOpts, &containerOwner)
+		if lastError != nil {
+			logrus.Error(lastError)
+		}
+		lastError = err
+	}
+	return lastError
+}
+
+func getUser(mountPoint string, userspec string) (specs.User, error) {
+	uid, gid, err := chrootuser.GetUser(mountPoint, userspec)
+	u := specs.User{
+		UID:      uid,
+		GID:      gid,
+		Username: userspec,
+	}
+	if !strings.Contains(userspec, ":") {
+		groups, err2 := chrootuser.GetAdditionalGroupsForUser(mountPoint, uint64(u.UID))
+		if err2 != nil {
+			if errors.Cause(err2) != chrootuser.ErrNoSuchUser && err == nil {
+				err = err2
+			}
+		} else {
+			u.AdditionalGids = groups
+		}
+
+	}
+	return u, err
+}
+
+func parsePath(runtime *libpod.Runtime, path string) (*libpod.Container, string) {
+	pathArr := strings.SplitN(path, ":", 2)
+	if len(pathArr) == 2 {
+		ctr, err := runtime.LookupContainer(pathArr[0])
+		if err == nil {
+			return ctr, pathArr[1]
+		}
+	}
+	return nil, path
+}
+
+func getPathInfo(path string) (string, os.FileInfo, error) {
+	path, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "error evaluating symlinks %q", path)
+	}
+	srcfi, err := os.Stat(path)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "error reading path %q", path)
+	}
+	return path, srcfi, nil
+}
+
+func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, chownOpts *idtools.IDPair) error {
+	srcPath, err := filepath.EvalSymlinks(src)
+	if err != nil {
+		return errors.Wrapf(err, "error evaluating symlinks %q", srcPath)
+	}
+
+	srcPath, srcfi, err := getPathInfo(srcPath)
+	if err != nil {
+		return err
+	}
+	if srcfi.IsDir() || dest[len(dest)-1] == os.PathSeparator {
+		if err = os.MkdirAll(destPath, 0755); err != nil {
+			return errors.Wrapf(err, "error creating directory %q", destPath)
+		}
+	} else {
+		if err = os.MkdirAll(filepath.Dir(destPath), 0755); err != nil {
+			return errors.Wrapf(err, "error creating directory %q", destPath)
+		}
+	}
+
+	copyFileWithTar := chrootarchive.CopyFileWithTarAndChown(chownOpts, digest.Canonical.Digester().Hash(), idMappingOpts.UIDMap, idMappingOpts.GIDMap)
```

Comment:
Maybe a comment here explaining that these return functions for copying items?

Reply:
fixed